### ISSUE TITLE
Poor documentation of default_content()

### ISSUE THREAD

**Issue by paran1** [voxpupuli/puppet-extlib#87]
Current documentation states that the function:
"Takes an optional content and an optional template name and returns the contents of a file."

It does not return "the contents of a file". It returns the first argument (if not empty or undef) otherwise it uses the second argument as the file name of a template, and returns the evaluated template. If both arguments are empty or undef, the function returns undef.

The example show that one intended use case is to use the result as the content attribute of a file type. Maybe this is why the documentation ended up like it is?

**Comment by ekohl** [comment]
You're right that the current documentation is poor. Would you be willing to submit a PR to improve it?

**Comment by paran1** [comment]
I usually try to contribute fixes. However, in this case you would likely get far better documentation from somebody who actually uses the function. I might be missing something, but currently I can't think of many use cases.

**Comment by alexjfisher** [comment]
It was some time ago, but perhaps the original author might be able to help with a sentence or two? @alvagante Sorry, I know it was 3 years ago! :)
